text/0000-9t_2b_encoding.md

```diff
@@ -0,0 +1,90 @@
+* **10/2**: `(243/10)*2 = 243/5 = 48.6 bytes`
+* **9/2**: `(243/ 9)*2 = 243/4.5 = 54 bytes`
+
+Additionally let's consider an IOTA transaction which usually consists of 8019 trits:
+* **10/2**: `(8019/10)*2 = 8019/5 = 1603,8 bytes`
+* **9/2**: `(8019/ 9)*2 = 8019/4.5 = 1782 bytes`
+
+There are three things noticeable here:
+* the `9/2` unsurprisingly requires more bytes to represent the same amount of information (is less memory efficient),
+* the `9/2` encoding maps onto integers for commonly used trit lengths, like the ones appearing as transaction field lengths: 27, 81, 243, 6561, 8019,
+* the more data to encode the bigger the gap between both encodings gets in terms of memory efficiency.
+
+Using the `9/2` encoding means additional - although redundant - `178,2` sent and stored bytes per uncompressed transaction. That might seem like a small overhead, but actually is quite significant considering billions of transactions going over the wire on a global scale.
+
+There are, however, some reasons that might mitigate the downsides quite a bit:
+* **trimming**: IOTA transactions, especially data transactions which do not fully utilize the available payload size, can be trimmed. The more a transaction can be trimmed the less overhead there is for the `9/2` encoding.
+* **steganography**: Since `3^9` is even smaller than `2^15 = 32768`, we could make use of an unused bit every two bytes to store some additional metadata. In case of an uncompressed transaction there would be `1782/2/8 = ~111` bytes available. This would reduce the memory overhead - in the best case scenario where those residual bits are fully used - to mere `66.825` bytes, which is almost a 3 times improvement.
+
+**Why are we doing this?**
+- **compression**: efficient trimming of zero bytes in IOTA transactions on the order of microseconds (tested on a Raspberry 3B+), ~10x faster than general purpose lz4 compresssion, which is therefore possibly insignificant compared to other more intense node operations in the processing pipeline, 
+- **(almost) no conversions**: since this encoding plays nicely with byte-wise data processing, bytes as received over the wire can be stored as-is in memory without any prior conversion involving memory allocations and copy operations to receive the converted representation 
+- **partial deserialization**: enables fast deserialization of single transaction fields by simply pointing into a single array with integer offsets and field lengths, thereby preventing costly memory allocations for each transaction field.
+
+# Detailed design
+
+TODO: FINISH THIS SECTION
+
+**Example transaction implementation based on 9/2 encoding**: This example assumes that a trivial trimming algorithm is used that only trims the 0 bytes from the signature message fragment.
```

Comment:
Really appreciate the example. We may also need to know what basic data structure looks like (maybe like #5 )